code/xcvm/cosmwasm/contracts/order/src/lib.rs

```diff
@@ -124,15 +124,18 @@ pub struct SolvedOrder {
 impl SolvedOrder {
 	pub fn new(order: OrderItem, solution: Cow) -> StdResult<Self> {
 		ensure!(
-			order.msg.wants.amount.u128() >= solution.given.0,
-			StdError::GenericErr { msg: "user limit was not satisfied".to_string() }
+			order.msg.wants.amount.u128() <= solution.given.u128(),
+			StdError::generic_err(format!(
+				"user limit was not satisfied {:?} {:?} ",
+				&order, &solution
```

Comment:
```suggestion
				"user limit was not satisfied {order:?} {solution:?} ",
```

Reply:
it uses references or clones? would it move objects into log. I am just not sure it compiles

Reply:
Clones are never done implicitly and `format` always works on references. If you use `&order` you’re effectively using a double reference. It compiles to the same thing of course but just using it in format message is more concise.